lib/merkle_tree.ex

```diff
@@ -42,35 +45,55 @@ defmodule MerkleTree do
     Alternatively, you can supply your own hash function that has the spec
     ``(String.t -> String.t)``.
   """
-  @spec new(blocks, hash_function) :: t
-  def new(blocks, hash_function \\ &MerkleTree.Crypto.sha256/1)
-  when blocks != [] do
-    unless is_power_of_n(@number_of_children, Enum.count(blocks)), do: raise MerkleTree.ArgumentError
-
-    root = build(blocks, hash_function)
+  @spec new(blocks, hash_function, height) :: t
+  def new(blocks, hash_function \\ &MerkleTree.Crypto.sha256/1, height \\ @default_height) do
```

Comment:
if length(blocks) > 2^height, then it enters never-ending loop

Reply:
done